sourcegraph/sourcegraph
# Aggregations: add extended timeout feature

Closes [Search aggregation insights do not allow to trigger explicit aggregation when timeout was hit](https://github.com/sourcegraph/sourcegraph/issues/40861)

[Design](https://www.figma.com/file/cKSeCtmBh9aiPsAVoKsLgM/Aggregation-insights%3A-display-%22grouped%22-insights-for-existing-searches-%2339126-%5Bready-for-dev%5D?node-id=2653%3A30114)

### Screenshot of implementation

![PixelSnap 2022-09-01 at 10 26 23](https://user-images.githubusercontent.com/1855233/187952591-14387763-2af4-4c73-9bf9-1677adf39b47.png)

### Screenshot of design

![PixelSnap 2022-09-01 at 10 27 21](https://user-images.githubusercontent.com/1855233/187952773-dcf18d52-9b9d-4067-9ce5-59174ba8dc54.png)

### Request timing out in 2 seconds then with 60 second max timeout

![PixelSnap 2022-09-01 at 10 14 26](https://user-images.githubusercontent.com/1855233/187949957-8ffc7eed-e9dd-45ac-b2f4-3619204d2579.png)

## Test plan

- Open devtools and switch to the "Network" tab
- Filter for "GetSearchAggregation"
- Run a search for any term. Use `type:diff` to make sure it times out (Should timeout in 2 seconds in the network tab)
- Should display "Run aggregation" over the skeleton chart as seen in the screenshot above
- Clicking button should re-run aggregation query with a 60s timeout
- Clicking another aggregation button should maintain the 60s timeout
- Changing the query should reset timeout to 2s

## App preview:

- [Web](https://sg-web-insights-aggregation-explicit.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-caxyngvfnb.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
